File: webui/src/main/java/com/taskadapter/webui/WebUserSession.java
package com.taskadapter.webui;

import java.util.Optional;

public class WebUserSession {

    Optional<String> currentLoginName = Optional.empty();

    public Optional<String> getCurrentUserName() {
        return currentLoginName;
    }

    public WebUserSession setCurrentUserName(String loginName) {
        this.currentLoginName = Optional.of(loginName);
        return this;
    }

    public void clear() {
        currentLoginName = Optional.empty();
    }
}

File: webui/src/main/scala/com/taskadapter/webui/WebUserSession.scala
package com.taskadapter.webui

class WebUserSession(val pageContainer: PageContainer) {

  var currentLoginName: Option[String] = None

  def getCurrentUserName: Option[String] = {
    currentLoginName
  }

  def setCurrentUserName(loginName: String): Unit = {
    currentLoginName = Some(loginName)
  }

  def clear(): Unit = {
    currentLoginName = None
  }
}
